p-token/src/processor/shared/burn.rs

```diff
@@ -78,10 +80,7 @@ pub fn process_burn(
     } else {
         source_account.set_amount(updated_source_amount);
 
-        let mint_supply = mint
-            .supply()
-            .checked_sub(amount)
-            .ok_or(TokenError::Overflow)?;
+        let mint_supply = mint.supply().checked_sub(amount).unwrap();
```

Comment:
Can you add a comment saying why this is ok? Also, what are the CU usage gains from unwrapping?

Reply:
Unwrapping is `2` CUs cheaper.